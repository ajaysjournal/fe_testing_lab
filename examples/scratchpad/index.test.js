import { test, expect, vi } from 'vitest';
// vi.spyOn is a function provided by the vitest testing framework
// that allows you to create a mock function that spies on a method of an object.
// This is useful for testing purposes, as it lets you track calls to the method,
// including the arguments passed and the return values.
const logSpy = vi.spyOn(console, 'log');

test('a super simple test', () => {
  console.log('Hello, world!');

  //
  expect(logSpy).toHaveBeenCalled();
  expect(logSpy).toHaveBeenCalledWith('Hello, world!');
  expect(logSpy).toHaveBeenCalledTimes(1);

});
