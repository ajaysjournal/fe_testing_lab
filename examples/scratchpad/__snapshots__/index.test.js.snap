// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`a super simple test 1`] = `"<div>wowowow</div>"`;
